pandas/core/window/indexers.py

```diff
@@ -0,0 +1,174 @@
+import abc
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+
+from pandas.tseries.offsets import DateOffset
+
+BeginEnd = Tuple[np.ndarray, np.ndarray]
+
+
+class BaseIndexer(abc.ABC):
+    """Base class for window bounds calculations"""
+    def __init__(self,
+                 index=None,
+                 offset: Optional[Union[str, DateOffset]] = None,
+                 keys: Optional[Sequence[np.ndarray]] = None):
+        """
+        Parameters
+        ----------
+        index : , default None
+            pandas index to reference in the window bound calculation
+
+        offset: str or DateOffset, default None
+            Offset used to calcuate the window boundary
+
+        keys: np.ndarray, default None
+            Additional columns needed to calculate the window bounds
+
+        """
+        self.index = index
+        self.offset = offset
+        self.keys = keys
+
+    @classmethod
+    @abc.abstractmethod
+    def get_window_bounds(cls,
+                          values: Optional[np.ndarray] = None,
+                          window_size: int = 0,
+                          min_periods: Optional[int] = None,
+                          center: Optional[bool] = None,
+                          closed: Optional[str] = None,
+                          win_type: Optional[str] = None) -> BeginEnd:
+        """
+        Computes the bounds of a window.
+
+        Parameters
+        ----------
+        # TODO: should users have access to _all_ the values or just the length (len(values))?
```

Comment:
The existing window calc routines only really need `len(values)`, but should users have access to the actual values?

Reply:
I don't think we need anything other than the index values for our use cases?